### ISSUE TITLE
unclear crash and error message

### ISSUE THREAD

**Issue by PaulSoderlind** [JuliaLang/julia#51792]
running a script in Julia 1.10.0-beta3, Win 64. Has happened several times. Does not happen in 1.09.
line 51 calls on XLSX, so maybe that's the problem?
 
```
GC error (probable corruption)                                                     
Allocations: 45444762 (Pool: 45321637; Big: 123125); GC: 19                        
                                                                                   
Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.                                        
Exception: EXCEPTION_ACCESS_VIOLATION at 0x7ffe539be538 -- jl_static_is_function_ at C:/workdir/src\rtutils.c:696 [inlined]                                           
jl_static_show_x_ at C:/workdir/src\rtutils.c:1113                                 
in expression starting at C:\Users\XXX\PSDataE\Classes\EmpFinPhD\EFPredGoyalWelch.jl:51                                                                        
Allocations: 45444762 (Pool: 45321637; Big: 123125); GC: 19   
```

**Comment by gbaraldi** [comment]
Can you share the code you use to reproduce this?

**Comment by PaulSoderlind** [comment]
I have tried create a small example, but failed. It seems as this happens only now and then - and after running the same code several times, sometimes with `error("stop here")` inserted somewhere. 

The line where the code block that seems to crash is this:
```
 XLSX.openxlsx(File) do fh
      data = isCA ? fh[sheet] : fh[sheet][:]
    end
```
where `isCA` is `false` in this run.

I did manage to capture a longer error message:
```
C error (probable corruption)                                                     
Allocations: 46658162 (Pool: 46532935; Big: 125227); GC: 20                        
                                                                                   
Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.                                        
Exception: EXCEPTION_ACCESS_VIOLATION at 0x7ffe89eae538 -- jl_static_is_function_ at C:/workdir/src\rtutils.c:696 [inlined]                                           
jl_static_show_x_ at C:/workdir/src\rtutils.c:1113                                 
in expression starting at C:\Users\XXX\PSDataE\Classes\EmpFinPhD\EFPredGoyalWelch.jl:51                                                                        
jl_static_is_function_ at C:/workdir/src\rtutils.c:696 [inlined]                   
jl_static_show_x_ at C:/workdir/src\rtutils.c:1113                                 
jl_static_show_next_ at C:/workdir/src\rtutils.c:1262                              
jl_static_show_x at C:/workdir/src\rtutils.c:1207 [inlined]                        
ijl_static_show at C:/workdir/src\rtutils.c:1268 [inlined]                         
jl_ at C:/workdir/src\rtutils.c:1358                                               
gc_dump_queue_and_abort at C:/workdir/src\gc.c:1830                                
gc_mark_outrefs at C:/workdir/src\gc.c:2547 [inlined]                              
gc_mark_and_steal at C:/workdir/src\gc.c:2765                                      
gc_mark_loop_parallel at C:/workdir/src\gc.c:2834                                  
gc_mark_loop at C:/workdir/src\gc.c:2855 [inlined]                                 
gc_mark_loop at C:/workdir/src\gc.c:2849 [inlined]                                 
_jl_gc_collect at C:/workdir/src\gc.c:3178                                         
ijl_gc_collect at C:/workdir/src\gc.c:3480                                         
gc at .\gcutils.jl:129 [inlined]                                                   
gc at .\gcutils.jl:129 [inlined]                                                   
#openxlsx#22 at C:\Users\XXX\.julia\packages\XLSX\we7i6\src\read.jl:151     
openxlsx at C:\Users\XXX\.julia\packages\XLSX\we7i6\src\read.jl:128 [inlined]                                                                                  
#Readxls3Ps#62 at C:\Users\XXX\PSDataD\PsCode\PsJulia\PaulS\Readxls3Ps.jl:65Readxls3Ps at C:\Users\XXX\PSDataD\PsCode\PsJulia\PaulS\Readxls3Ps.jl:47    
unknown function (ip: 000002d69cbc5c57)                                            
jl_apply at C:/workdir/src\julia.h:1976 [inlined]                                  
do_call at C:/workdir/src\interpreter.c:125                                        
eval_value at C:/workdir/src\interpreter.c:222                                     
eval_stmt_value at C:/workdir/src\interpreter.c:173 [inlined]                      
eval_body at C:/workdir/src\interpreter.c:634                                      
jl_interpret_toplevel_thunk at C:/workdir/src\interpreter.c:774                    
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:934                             
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:877                             
ijl_toplevel_eval at C:/workdir/src\toplevel.c:943 [inlined]                       
ijl_toplevel_eval_in at C:/workdir/src\toplevel.c:985                              
eval at .\boot.jl:383 [inlined]                                                    
include_string at .\loading.jl:2070                                                
_include at .\loading.jl:2130                                                      
include at .\client.jl:489                                                         
unknown function (ip: 000002d6f7ca0eab)                                            
jl_apply at C:/workdir/src\julia.h:1976 [inlined]                                  
do_call at C:/workdir/src\interpreter.c:125                                        
eval_value at C:/workdir/src\interpreter.c:222                                     
eval_stmt_value at C:/workdir/src\interpreter.c:173 [inlined]                      
eval_body at C:/workdir/src\interpreter.c:634                                      
jl_interpret_toplevel_thunk at C:/workdir/src\interpreter.c:774                    
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:934                             
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:877                             
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:877                             
ijl_toplevel_eval at C:/workdir/src\toplevel.c:943 [inlined]                       
ijl_toplevel_eval_in at C:/workdir/src\toplevel.c:985                              
eval at .\boot.jl:383 [inlined]                                                    
eval_user_input at C:\workdir\usr\share\julia\stdlib\v1.10\REPL\src\REPL.jl:150    
repl_backend_loop at C:\workdir\usr\share\julia\stdlib\v1.10\REPL\src\REPL.jl:246  
#start_repl_backend#46 at C:\workdir\usr\share\julia\stdlib\v1.10\REPL\src\REPL.jl:231                                                                                
start_repl_backend at C:\workdir\usr\share\julia\stdlib\v1.10\REPL\src\REPL.jl:228 
#run_repl#59 at C:\workdir\usr\share\julia\stdlib\v1.10\REPL\src\REPL.jl:389       
run_repl at C:\workdir\usr\share\julia\stdlib\v1.10\REPL\src\REPL.jl:375           
jfptr_run_repl_95404.1 at C:\Program Files\Julia\Julia-1.10\lib\julia\sys.dll (unknown line)                                                                          
#1013 at .\client.jl:432                                                           
jfptr_YY.1013_86251.1 at C:\Program Files\Julia\Julia-1.10\lib\julia\sys.dll (unknown line)                                                                           
jl_apply at C:/workdir/src\julia.h:1976 [inlined]                                  
jl_f__call_latest at C:/workdir/src\builtins.c:812                                 
#invokelatest#2 at .\essentials.jl:887 [inlined]                                   
invokelatest at .\essentials.jl:884 [inlined]                                      
run_main_repl at .\client.jl:416                                                   
exec_options at .\client.jl:333                                                    
_start at .\client.jl:552                                                          
jfptr__start_86276.1 at C:\Program Files\Julia\Julia-1.10\lib\julia\sys.dll (unknown line)                                                                            
jl_apply at C:/workdir/src\julia.h:1976 [inlined]                                  
true_main at C:/workdir/src\jlapi.c:582                                            
jl_repl_entrypoint at C:/workdir/src\jlapi.c:731                                   
mainCRTStartup at C:/workdir/cli\loader_exe.c:58                                   
BaseThreadInitThunk at C:\windows\System32\KERNEL32.DLL (unknown line)             
RtlUserThreadStart at C:\windows\SYSTEM32\ntdll.dll (unknown line)                 
Allocations: 46658162 (Pool: 46532935; Big: 125227); GC: 20                        
```

**Comment by PaulSoderlind** [comment]
Seems fixed in 1.10.0-rc3. Cannot reproduce it any more.
